src/follower.py

```diff
@@ -106,11 +113,71 @@ def stop(self):
 
 
     def turn(self):
-        """Turn the wheels towards the last recognized object."""
+        """
+        Turn the wheels towards the last recognized object.
+        """
         self.convert_camera_angle()
-        self._fw.turn(self._turn_angle)
+        self.pan_camera()
+
+        if self.STRAIGHT_ANGLE + self.DEADZONE_WHEELS < self._turn_angle:
+            # Left condition: when the turn angle is between 90 + deadzone and 135, turn the wheels at that angle plus the offset. 
+            # If the turn angle is over the maximum wheel turn angle, set the turn angle to the wheel max.
+            self._fw.turn(self._turn_angle + self.camera_angle_offset) 
+            if self._turn_angle + self.camera_angle_offset > self.WHEEL_MAX:
+              self._fw.turn(self.WHEEL_MAX)
+
+        elif self.STRAIGHT_ANGLE - self.DEADZONE_WHEELS > self._turn_angle:
+            # Right condition: when the turn angle is between 90 - deadzone and 45, turn the wheels at that angle minus the offset. 
+            # If the turn angle is under the minimum wheel turn angle, set the turn angle to the wheel min.
+            self._fw.turn(self._turn_angle - self.camera_angle_offset)
+            if self._turn_angle - self.camera_angle_offset < self.WHEEL_MIN:
+              self._fw.turn(self.WHEEL_MIN)
+
+
+    def pan_camera(self):
+        """
+        Follows the tag by panning camera towards the same direction as the wheels.
+        """
+        if self._turn_angle < self.STRAIGHT_ANGLE - self.DEADZONE_CAMERA:
+            # Left condition: If the turn angle is less than 90 - deadzone, turn the camera left. 
+            # The offset is 90 - the current angle.
+            self.turn_camera_left(self._turn_angle)
+            self.camera_angle_offset = np.abs(90 - self._camera.current_pan)
+
+        elif self._turn_angle > self.STRAIGHT_ANGLE + self.DEADZONE_CAMERA:
+            # Right condition: If the turn angle is greater than 90 + deadzone, turn the camera right. 
+            # The offset is the current angle - 90.
+            self.turn_camera_right(self._turn_angle)
+            self.camera_angle_offset = np.abs(self._camera.current_pan - 90)
+
+        if self.camera_angle_offset == self.STRAIGHT_ANGLE or self.camera_angle_offset == self.WHEEL_MIN:
+            # If the offset of the right or left offsets are equal to 90 or 45 respectively, turn the camera straight since the wheels will be straight.
+            self._camera.current_pan = self.STRAIGHT_ANGLE
+            self._camera.pan_servo.write(self._camera.current_pan)
+            self._fw.turn(self._camera.current_pan)
```

Comment:
Why aren't you updating the camera offset at the end? It will still use the old camera offset for at least one cycle.